server/src/main/java/org/elasticsearch/common/unit/ByteSizeValue.java

```diff
@@ -332,4 +332,22 @@ public int compareTo(ByteSizeValue other) {
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         return builder.value(toString());
     }
+
+    /**
+     * @return Constructs a {@link ByteSizeValue} with the bytes resulting from the addition of the arguments' bytes. Note that the
+     *         resulting {@link ByteSizeUnit} is bytes.
+     * @throws ArithmeticException if the resulting bytes overflow a long
+     */
+    public static ByteSizeValue add(ByteSizeValue x, ByteSizeValue y) {
+        return ByteSizeValue.ofBytes(Math.addExact(x.getBytes(), y.getBytes()));
```

Comment:
Can we throw if `x.getBytes() == -1 || y.getBytes() == -1`?

Reply:
Yes, will be in next push.